File: backend/functions/Period.py
from flask_sqlalchemy import SQLAlchemy
from flask import Flask
from .extensions import db
import pandas as pd


class Period(db.Model):
    __tablename__ = 'period'

    id = db.Column(db.Integer, primary_key=True)
    start_time = db.Column(db.Time, nullable=False)
    end_time = db.Column(db.Time, nullable=False)
    day = db.Column(db.Enum('MW', 'TR'))

    def __repr__(self):
        return '<Time Period %r %r %r>' % (self.day, self.start_time, self.end_time)
    
def load_time_data(csv_file):
    """
    Load time block data from a CSV file.

    Args:
    - csv_file (str): Path to the CSV file containing course data.

    Returns:
    - Tuple: A tuple containing three lists: the start time, end time, and what day
    """
    timeList = pd.read_csv(csv_file, dtype={0: 'str', 1: 'str'})
    startTime = timeList['Start Time'].tolist()
    endTime = timeList['End Time'].tolist()

    return startTime, endTime

File: frontend/src/app/_types/Period.ts
/**
 * Period
 * @uuid Unique ID for course object;
 * @start_time Start time for period;
 * @end_time End time for period;
 * @day Day enum for which day the period applies to;
 */
export type Period = {
  uuid: string;
  start_time: string;
  end_time: string;
  day: Day | "No Period Assigned";
};

/**
 * Day
 * @MW - Monday/Wednesday class
 * @TR - Tuesday/Thursday class
 */
export enum Day {
  MW = "MW",
  TR = "TR",
}

export const defaultPeriod = {
  start_time: "",
  end_time: "",
  day: Day["MW"],
};

export type ExportPeriod = {
  "Start Time": string;
  "End Time": string;
};